hmp/models/forward.py

```diff
@@ -0,0 +1,295 @@
+"""Models to estimate event probabilities."""
+
+import gc
+import itertools
+import multiprocessing as mp
+from itertools import cycle, product
+from warnings import resetwarnings, warn
+
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from pandas import MultiIndex
+from scipy.signal import correlate
+from scipy.stats import norm as norm_pval
+
+from hmp.models.base import BaseModel
+from hmp.models.fixedn import FixedEventModel
+
+try:
+    __IPYTHON__
+    from tqdm.notebook import tqdm
+except NameError:
+    from tqdm import tqdm
+
+default_colors = ["cornflowerblue", "indianred", "orange", "darkblue", "darkgreen", "gold", "brown"]
+
+
+class ForwardEstimationModel(BaseModel):
```

Comment:
Cumulative is probably a better as forward can refer to the forward variable computed during expectation-maximization
```suggestion
class CumulativeEstimationModel(BaseModel):
```